### PR TITLE
V2 filters frontend

### PR DESCRIPTION
**Pull request by RamtinTJB**
## Tracking Info

Resolves parts of #80 (Doesn't send the request to the backend on click)

## Changes

- Refactored old components
- Deleted old unused filter components
- Wrote reusable components for checkbox filters and radio button filters
- Made the NavBar sticky and implemented the frontend for the filter panel

## Testing

Since the "Apply" button doesn't actually send the request to the backend, the only way to test it to make sure it visually looks the same as the designs.

## Confirmation of Change

image

### CODE REVIEW

File: frontend/src/components/CheckboxFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+export const CheckboxFilter = (props: CheckboxFilterProps) => {
+  return (
+    <FilterContainer>
+      <FilterHeader title={props.title} />
+      {props.options.map((text, idx) => (
+        <CheckboxFilterRow key={idx}>
+          <FilterRadioButton
+            src={
+              props.value.has(idx) ? "/filled_filter_radio_button.svg" : "/filter_radio_button.svg"
```
Comment (petabite) [inline frontend/src/components/CheckboxFilter.tsx]: Would it be helpful to reuse the `CustomCheckboxRadio` from [here](https://github.com/TritonSE/USHS-Housing-Portal/blob/44c825e17a329b90484e7ee33328eb7370e11f86/frontend/src/components/ListingForm/CommonStyles.tsx#L68-L72)? Or do you still want to import the SVG for the checkbox?

**Review by petabite — CHANGES_REQUESTED** [review]
This looks great so far! 💯

Some things I noticed visually (see video below):
- The navbar moves when scrolling and gets hidden under the filter panel
- The pending/available unit toggle z-index is higher than the navbar

https://github.com/TritonSE/USHS-Housing-Portal/assets/24444266/dfb0e668-7b69-4694-82cf-d29dfc0f5a24

File: frontend/src/components/CheckboxFilter.tsx (hunk 1 above)
Comment (RamtinTJB) [reply frontend/src/components/CheckboxFilter.tsx]: Good point, I will reuse that. I didn't know that component existed :sweat_smile:

**Review by petabite — APPROVED** [review]
Nice! This looks really good!

**Review by soma-p — APPROVED** [review]
Looks great!

**Pull request merged**